### ISSUE TITLE
Wrong implementations of `get_preds()`

### ISSUE THREAD

**Issue by heyrict** [fastai/fastai#906]
**Describe the bug**.
`get_preds(is_test=True)` returns `[val_matrices, true_label]` instead of predicted labels.
It seems that `validate()` --&gt; `loss_batch()` does not returns any output.

**Expected behavior**
Add return `out` from `loss_batch()` from `validate()` and calculate true labels in get_preds,
or use another function to get predicted labels.

**Comment by sgugger** [comment]
Not sure I follow. `get_preds(is_test=True)` returns `[outputs, fake_label]` where the `fake_label` are what was given with the test set (usually a bunch of zeros) and `outputs` the outputs of the model on the test set. Depending on your loss function, you then have to apply a softmax/sigmoid to get your predictions from the outputs, but I don't see the bug.

**Comment by heyrict** [comment]
Thank you for your reply.
Maybe I'm misreading the code, but I'm not sure if `get_preds` returns the outputs.

In `get_preds`, we call `validate`; in validate, we call `loss_batch` and returns `val_score`

**Comment by sgugger** [comment]
`validate` then `loss_batch` are called without a `loss_fn` so we stop at [this line](https://github.com/fastai/fastai/blob/ba616ed2b8e132e446fb88d7d8acff7a38dd04e0/fastai/basic_train.py#L22) by returning `out` and `y`.

Closing this here as questions about the code should be asked on the [forum](https://forums.fast.ai/).
